File: models/assetmanagement/AssetCategory.js
import mongoose from "mongoose";

const AssetCategorySchema = new mongoose.Schema(
{
    hospital_id: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branch_id: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    name: { type: String, required: true },
    description: String,
    is_active: { type: Boolean, default: true }

},
{ timestamps: true }
);

export default mongoose.model("AssetCategory", AssetCategorySchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AssetCategory.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "asset_categories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AssetCategory extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @Column(nullable = false)
    private String name;

    private String description;

    @Column(name = "is_active")
    private boolean isActive = true;
}
